dace/transformation/dataflow/redundant_array.py

```diff
@@ -2,13 +2,84 @@
 """ Contains classes that implement a redundant array removal transformation.
 """
 
+import dace
+import functools
+import typing
+
+from copy import deepcopy as dcpy
 from dace import registry, subsets
 from dace.sdfg import nodes
 from dace.sdfg import utils as sdutil
+from dace.sdfg import graph
 from dace.transformation import transformation as pm
 from dace.config import Config
 
 
+# Helper methods #############################################################
+
+def _validate_subsets(edge: graph.MultiConnectorEdge,
+                      arrays: typing.Dict[str, dace.data.Data],
+                      src_name: str = None,
+                      dst_name: str = None) -> typing.Tuple[subsets.Subset]:
+    """ Extracts and validates src and dst subsets from the edge. """
+
+    dataname = edge.data.data
+
+    # Find src and dst names
+    if not src_name and isinstance(edge.src, nodes.AccessNode):
+        src_name = edge.src.data
+    if not dst_name and isinstance(edge.dst, nodes.AccessNode):
+        dst_name = edge.dst.data
+
+    # Find the src and dst subsets (deep-copy to allow manipulation)
+    if src_name == dataname:
+        src_subset = dcpy(edge.data.subset)
+        dst_subset = dcpy(edge.data.other_subset)
+    elif dst_name == dataname:
+        src_subset = dcpy(edge.data.other_subset)
+        dst_subset = dcpy(edge.data.subset)
+    else:
+        raise NotImplementedError
```

Comment:
I thought `edge.data.{src,dst}_subset` does the same?

Reply:
Is this being automatically set now? I still live in the old days ...

Reply:
Let's hope it works fine with everything :)